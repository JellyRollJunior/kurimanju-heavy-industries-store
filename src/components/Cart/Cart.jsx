import styles from './Cart.module.css';
import { productData } from '../../data/productData.js';
import { CartInfo } from '../CartInfo/CartInfo.jsx';
import { CartItem } from '../CartItem/CartItem.jsx';

const Cart = ({
  cart,
  hideCart,
  checkoutCart,
  removeFromCart,
}) => {
  const numCartItems = cart.reduce((total, quantity) => (total += quantity), 0);
  const subtotal = cart.reduce((accumulator, quantity, index) => {
    const itemPrice = Number(
      productData
        .find((product) => product.id == index)
        .price.replace(/\s+/g, '')
    );
    accumulator += itemPrice * quantity;
    return accumulator;
  }, 0);

  return (
    <aside className={styles.cart}>
      <section className={styles.titleHolder}>
        <h2>Cart</h2>
        <button onClick={hideCart} className={styles.hideCartButton}>
          &times;
        </button>
      </section>
      <CartInfo
        numCartItems={numCartItems}
        subtotal={subtotal}
        checkoutCart={checkoutCart}
      />
      <section>
        <ul>
          {cart.map(
            (quantity, itemId) =>
              quantity > 0 && (
                <CartItem
                  key={itemId}
                  itemId={itemId}
                  quantity={quantity}
                  removeFromCart={removeFromCart}
                />
              )
          )}
        </ul>
      </section>
    </aside>
  );
};

export { Cart };
