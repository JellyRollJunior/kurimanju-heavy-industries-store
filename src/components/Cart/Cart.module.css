.cartOverlay {
  display: flex;
  justify-content: end;
  padding: 65px 0;

  position: fixed;
  right: 0;
  height: 100vh;
  width: 100vw;
  background-color: #00000080;
}

.cart {
  composes: primaryBackground from '../../styles/colors.module.css';
  composes: quaternaryColor from '../../styles/colors.module.css';
  min-width: 250px;
  max-width: 95vw;
  padding: 8px 16px 0 12px;

  text-align: left;
  border-top: 2px solid;
  border-left: 2px solid;
  border-bottom: 2px solid;
  border-top-left-radius: 4px;
  border-bottom-left-radius: 4px;
  overflow-y: auto;
}

.titleHolder {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.hideCartButton {
  composes: quaternaryColor from '../../styles/colors.module.css';
  padding-right: 0;
  font-size: 30px;
  background: none;
  border: none;
}

ul {
  margin: 16px 0 14px 0;
  padding: 0;
}
