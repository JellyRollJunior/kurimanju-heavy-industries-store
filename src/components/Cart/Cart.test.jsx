import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import { Cart } from "./Cart";

describe('Shopping cart slide component', () => {
    it('Renders title, cart close button, checkout button, and cart item holder', () => {
        const cart = [1, 1, 1, 1, 1];
        const { container } = render(<Cart cart={cart} />);

        expect(container).toMatchSnapshot();
    })
})