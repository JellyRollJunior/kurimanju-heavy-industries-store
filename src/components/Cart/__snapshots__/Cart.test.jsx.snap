// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Shopping cart slide component > Renders title, cart close button, checkout button, and cart item holder 1`] = `
<div>
  <aside
    class="_cart_f8af68"
  >
    <section
      class="_titleHolder_f8af68"
    >
      <h2>
        Cart
      </h2>
      <button
        class="_hideCartButton_f8af68"
      >
        Ã—
      </button>
    </section>
    <section>
      <h3
        class="_subtotal_2d26a7"
      >
        Subtotal (
        5
         items):
      </h3>
      <h3
        class="_subtotal_2d26a7"
      >
        $ 
        500 000 000
      </h3>
      <button
        class="_checkoutButton_2d26a7"
      >
        Checkout
      </button>
    </section>
    <section>
      <ul>
        <li
          class="_cartItem_61ad6d"
        >
          <img
            alt="X-4 Base"
            src="/src/assets/X4_base.png"
          />
          <div
            class="_data_61ad6d"
          >
            <h3
              class="_name_61ad6d"
            >
              X-4 Base
            </h3>
            <h4
              class="_price_61ad6d"
            >
              90 000 000
            </h4>
            <h4
              class="_quantity_61ad6d"
            >
              Quantity: 
              1
            </h4>
            <button
              class="_removeButton_61ad6d"
            >
              Remove
            </button>
          </div>
        </li>
        <li
          class="_cartItem_61ad6d"
        >
          <img
            alt="X-4 Stealth"
            src="/src/assets/X4_stealth.png"
          />
          <div
            class="_data_61ad6d"
          >
            <h3
              class="_name_61ad6d"
            >
              X-4 Stealth
            </h3>
            <h4
              class="_price_61ad6d"
            >
              95 000 000
            </h4>
            <h4
              class="_quantity_61ad6d"
            >
              Quantity: 
              1
            </h4>
            <button
              class="_removeButton_61ad6d"
            >
              Remove
            </button>
          </div>
        </li>
        <li
          class="_cartItem_61ad6d"
        >
          <img
            alt="X-4 Respond"
            src="/src/assets/X4_respond.png"
          />
          <div
            class="_data_61ad6d"
          >
            <h3
              class="_name_61ad6d"
            >
              X-4 Respond
            </h3>
            <h4
              class="_price_61ad6d"
            >
              95 000 000
            </h4>
            <h4
              class="_quantity_61ad6d"
            >
              Quantity: 
              1
            </h4>
            <button
              class="_removeButton_61ad6d"
            >
              Remove
            </button>
          </div>
        </li>
        <li
          class="_cartItem_61ad6d"
        >
          <img
            alt="X-4 Enforce"
            src="/src/assets/X4_enforce.png"
          />
          <div
            class="_data_61ad6d"
          >
            <h3
              class="_name_61ad6d"
            >
              X-4 Enforce
            </h3>
            <h4
              class="_price_61ad6d"
            >
              100 000 000
            </h4>
            <h4
              class="_quantity_61ad6d"
            >
              Quantity: 
              1
            </h4>
            <button
              class="_removeButton_61ad6d"
            >
              Remove
            </button>
          </div>
        </li>
        <li
          class="_cartItem_61ad6d"
        >
          <img
            alt="X-4 Supress"
            src="/src/assets/X4_supress.png"
          />
          <div
            class="_data_61ad6d"
          >
            <h3
              class="_name_61ad6d"
            >
              X-4 Supress
            </h3>
            <h4
              class="_price_61ad6d"
            >
              120 000 000
            </h4>
            <h4
              class="_quantity_61ad6d"
            >
              Quantity: 
              1
            </h4>
            <button
              class="_removeButton_61ad6d"
            >
              Remove
            </button>
          </div>
        </li>
      </ul>
    </section>
  </aside>
</div>
`;
